pgtoolkit/log/syslog.py

```diff
@@ -29,14 +49,22 @@ def parse(self, lines):
             m = self._re.match(line)
             if m:
                 message = line[len(m.group(0)):]
-                yield SyslogLine(message, **m.groupdict())
+                yield SyslogLine(
+                    message, date_parser=self.date_parser, **m.groupdict())
             else:
                 yield line
 
 
 class SyslogLine(UserString):
-    def __init__(self, message, **kw):
+    def __init__(self, message, date_parser=None, **kw):
         if message.startswith('#011'):
             message = '\t' + message[4:]
         self.__dict__.update(kw)
         super(SyslogLine, self).__init__(message)
+        self.date_parser = date_parser or parse_syslog_datetime
+
+    def parse_stage2(self, record):
+        if not hasattr(record, 'pid'):
+            record.pid = int(self.pid)
```

Comment:
this looks unrelated to date parsing, perhaps move into a dedicated commit (along with respective tests)?